File: backend/internal/core/domain/Stage.go
package domain

import "go.mongodb.org/mongo-driver/bson/primitive"

// swagger:model Stage
type Stage struct {
	ID     primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
	Name   string             `bson:"name" json:"name"`
	Active bool               `bson:"active" json:"active"`
}

File: frontend/src/app/types/stage.ts
export interface StageDto {
    name: string;
}

export interface Stage {
    id: string;
    name: string;
    active: boolean;
}

export function createMockStage(): Stage {
    return {
        active: true,
        id: '1',
        name: 'name',
    };
}
